### PR TITLE
Refactor tftp handling

### PR DESCRIPTION
**Pull request by ekohl**
Uses modern puppet features to simpify the code.

### CODE REVIEW

**Comment by domcleal** [comment]
Could you check and fix the test failures please?

**Comment by ekohl** [comment]
Now with passing tests.

File: spec/classes/foreman_proxy__tftp_spec.rb [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        facts
+      end
+
+      let :pre_condition do
+        'include ::foreman_proxy'
+      end
+
+      it { is_expected.to compile.with_all_deps }
+
+        tftp_root = case facts[:osfamily]
```
Comment (domcleal) [inline spec/classes/foreman_proxy__tftp_spec.rb]: Nitpick, indentation is off here.

File: spec/classes/foreman_proxy__tftp_spec.rb [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+      elsif facts[:osfamily] == 'RedHat'
+        if facts[:operatingsystemmajrelease].to_i > 6
+          it { should contain_package('grub2-efi').with_ensure('installed') }
+          it { should contain_package('grub2-efi-modules').with_ensure('installed') }
+          it { should contain_package('grub2-tools').with_ensure('installed') }
+          it { should contain_package('shim').with_ensure('installed') }
+
+          case facts[:operatingsystem]
+          when /^(RedHat|Scientific|OracleLinux)$/
+              it { should contain_file("#{tftp_root}/grub2/grubx64.efi").with_source('/boot/efi/EFI/redhat/grubx64.efi') }
```
Comment (domcleal) [inline spec/classes/foreman_proxy__tftp_spec.rb]: Nitpick, indentation here too (4, not 2 spaces)

File: manifests/tftp.pp [hunk 3]
```diff
@@ -1,50 +1,64 @@
...
-    ensure_packages(['wget'], { ensure => $foreman_proxy::ensure_packages_version, })
+  if $manage_wget {
+    ensure_packages(['wget'], { ensure => $wget_version, })
   }
 
   case $::osfamily {
     'RedHat': {
-      $osreleasemajor = regsubst($::operatingsystemrelease, '^(\d+)\..*$', '\1') # workaround for the possibly missing operatingsystemmajrelease
-      if versioncmp($osreleasemajor, '6') <= 0 {
+      if versioncmp($::operatingsystemmajrelease, '6') <= 0 {
```
Comment (domcleal) [inline manifests/tftp.pp]: Nitpick, you can rely on `$facts['operatingsystemmajrelease']` or even `$facts['os']['release']['major']` if you want to completely modernise this for Puppet 4.

**Comment by mmoll** [comment]
merged with stylefixes, bedankt @ekohl!

**Pull request merged**